code-challenges.js

```diff
@@ -21,12 +32,30 @@ var testingArray1 = [23, "Heyyyy!", 45, -9, 0, "Yo", false]
 var testingArray2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
-
-
+// Start by creating a function that takes in array of mixed data. 
+const gimmeTheRemainder = (array) => {
+// // We are probably going to filter first in order to only extract the numbers from the mixed data array.  
+  let onlyNumbs = array.filter(numbers => typeof numbers == "number")
+  return onlyNumbs.map(numbers => numbers %3)
+// Now we want to map at over our array of numbers and use %3 to get the remainder of each number.
+}
+ console.log(gimmeTheRemainder(testingArray1))
+ console.log(gimmeTheRemainder(testingArray2))
 
 
 // --------------------3) Create a function that takes in two arrays as arguments returns one array with no duplicate values.
 
 var testingArray3 = [3, 7, "hi", 10, 3, "hello", 4, "hi"]
 var testingArray4 = [7, "hi", 3, 1, "hi", 4, "hello", 4, 7]
 // Expected output: [ 3, 7, "hi", 10, "hello", 4, 1 ]
+
+// Start by creating a function that takes in two arrays and returns one array with no duplicates
+
+const noDuples = (array1, array2) => {
+  // Now create a local variable that can hold our new array with no duplicates.
+  // Use concat to merge both arrays.
+  let array3 = array1.concat(array2)
+  return [...new Set(array3)]
```

Comment:
I'm impressed with your use of the spread operator here. I used filter but this seems so much more "badass" seeing how it's just one line. well done!

Reply:
why is the use of Set necessary here?